plugins/serializers/binary/binary.go

```diff
@@ -0,0 +1,119 @@
+package binary
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"log"
+)
+
+type Serializer struct {
+	Entries    []Entry `toml:"entries"`
+	Endianness string  `toml:"endianness"`
+
+	converter binary.ByteOrder
+}
+
+func (s *Serializer) Init() error {
+	log.Printf("I [binary] binary serializer init")
+
+	if s.Endianness == "" {
+		s.Endianness = "host"
+	}
+
+	switch s.Endianness {
+	case "big":
+		s.converter = binary.BigEndian
+	case "little":
+		s.converter = binary.LittleEndian
+	case "host":
+		s.converter = internal.HostEndianness
+	default:
+		return fmt.Errorf("invalid endianness %q", s.Endianness)
+	}
+
+	for i, entry := range s.Entries {
+		if err := entry.Check(); err != nil {
+			return err
```

Comment:
Maybe also inform the user which entry had the error?

Reply:
Sure, should be added with the latest commit